# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [3]:
a = np.random.randint(1,5,size=100)
b = np.random.randint(1,4,size=100)
c = np.random.randint(10000,30001,size=100)

### Step 3. Let's create a DataFrame by joinning the Series by column

In [6]:
df = pd.DataFrame(data=[a,b,c]).T

In [7]:
df

0  1      2
0   3  2  17158
1   3  3  26026
2   3  1  18680
3   3  1  11526
4   4  2  18546
.. .. ..    ...
95  3  2  15617
96  2  2  11239
97  2  1  18375
98  1  2  26874
99  3  3  24113

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [9]:
columnas = ['bedrs','bathrs','price_sqr_meter']

In [11]:
df.rename(columns={0:'bedrs',1:'bathrs',2:'price_sqr_meter'})

bedrs  bathrs  price_sqr_meter
0       3       2            17158
1       3       3            26026
2       3       1            18680
3       3       1            11526
4       4       2            18546
..    ...     ...              ...
95      3       2            15617
96      2       2            11239
97      2       1            18375
98      1       2            26874
99      3       3            24113

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [19]:
arr = df.values.copy()

In [21]:
arr.resize(300,1)

In [22]:
pd.DataFrame(arr)

0
0        3
1        2
2    17158
3        3
4        3
..     ...
295      2
296  26874
297      3
298      3
299  24113

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

### Step 7. Reindex the DataFrame so it goes from 0 to 299